_examples/checkout-branch/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/go-git/go-git/v5"
+	. "github.com/go-git/go-git/v5/_examples"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+)
+
+// Checkout a branch
+func main() {
+	CheckArgs("<url>", "<directory>", "<branch>")
+	url, directory, branch := os.Args[1], os.Args[2], os.Args[3]
+
+	// Clone the given repository to the given directory
+	Info("git clone %s %s", url, directory)
+	r, err := git.PlainClone(directory, false, &git.CloneOptions{
+		URL: url,
+	})
+	CheckIfError(err)
+
+	// ... retrieving the commit being pointed by HEAD
+	Info("git show-ref --head HEAD")
+	ref, err := r.Head()
+	CheckIfError(err)
+
+	fmt.Println(ref.Hash())
+
+	w, err := r.Worktree()
+	CheckIfError(err)
+
+	// ... checking out branch
+	Info("git checkout %s", branch)
+
+	branchRefName := fmt.Sprintf("refs/heads/%s", branch)
```

Comment:
```suggestion
	branchRefName := plumbing.NewBranchReferenceName(branch)
```

Reply:
Great - thanks - used.  Sorry, didn't see this.